src/arrayMethodJoin.js

```diff
@@ -4,8 +4,21 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      const current = this[i] === null || this[i] === undefined ? '' : this[i];
+
+      result += current;
```

Comment:
```suggestion
      if (this[i] !== null && this[i] !== undefined) {
         result += current;
      }
```
as for me this version looks more readable